scripts/get_maintainer.py

```diff
@@ -224,11 +228,29 @@ def download(pr):
     return f.name
 
 
+def show_release_to(subsystems):
```

Comment:
Works, but there is an easier way to achieve the same, see the diff below. Also, with `sorted(...)` around the `set(...)` you'll get the same order every time calling this script instead of random order. I leave it up to you to decide which version you prefer. In any case
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`
```diff
diff --git a/scripts/get_maintainer.py b/scripts/get_maintainer.py
index 7006dca2..867e6d5a 100755
--- a/scripts/get_maintainer.py
+++ b/scripts/get_maintainer.py
@@ -228,17 +228,10 @@ def download(pr):
     return f.name
 
 
-def show_release_to(subsystems):
-    to = set()
-    for ss in subsystems:
-        for tag in subsystems[ss]:
-            if tag == 'R' or tag == 'M':
-                for name in subsystems[ss][tag]:
-                    # Remove eventual GitHub handle
-                    name = re.sub(r" *\[@.*\]$", '', name)
-                    if name:
-                        to.add(name)
-    print(*to, sep=', ')
+def show_release_to():
+    with open("MAINTAINERS", "r") as f:
+        emails = sorted(set(re.findall(r'[RM]:\t(.*[\w]*<[\w\.-]+@[\w\.-]+>)', f.read())))
+        print(*emails, sep=', ')
 
 
 def main():
@@ -248,7 +241,7 @@ def main():
     all_subsystems = parse_maintainers()
 
     if args.release_to:
-        show_release_to(all_subsystems)
+        show_release_to()
         return
 
     paths = []
```

Reply:
Looks good :) Another advantage with your solution is, no need to do the full `parse_maintainers()` first (although the path check would still be needed). I'll update the PR.

Reply:
Speaking about how we're checking the path in the script. That seems like a rather complicated way of checking that we're running it from the root folder? This would also achieve the same.
```python
diff --git a/scripts/get_maintainer.py b/scripts/get_maintainer.py
index 3431fea7..179a9d1c 100755
--- a/scripts/get_maintainer.py
+++ b/scripts/get_maintainer.py
@@ -54,9 +54,7 @@ def get_args():
 
 
 def check_cwd():
-    cwd = os.getcwd()
-    parent = os.path.dirname(os.path.realpath(__file__)) + "/../"
-    if (os.path.realpath(cwd) != os.path.realpath(parent)):
+    if not os.path.exists(os.getcwd() + "/MAINTAINERS"):
         print("Error: this script must be run from the top-level of the "
               "optee_os tree")
         exit(1)

```

Reply:
@jbech-linaro haha, funny, it is the opposite situation we had https://github.com/OP-TEE/optee_os/pull/3193#discussion_r312841651 where I proposed to check the existence of MAINTAINERS in the current directory, and you proposed to check the script's parent :laughing:

In the end I think the current solution is more robust because it would not break if run from some place that happens to have a MAINTAINERS file.

Reply:
I think my proposal in that comment was about being able to run it from anywhere, i.e, root folder, script-folder, $HOME whatever. But not a big deal, the current way works.

> In the end I think the current solution is more robust because it would not break if run from some place that happens to have a MAINTAINERS file.

And that would only happen when the sun is at 32 degrees elevation to the earth and the Atlantic ocean have dried up :smile:.